FFMpegCore/FFMpeg/FFMpeg.cs

```diff
@@ -116,7 +116,12 @@ await arguments
         {
             captureTime ??= TimeSpan.FromSeconds(source.Duration.TotalSeconds / 3);
             size = PrepareSnapshotSize(source, size);
-            streamIndex = streamIndex == null ? 0 : source.VideoStreams.FirstOrDefault(videoStream => videoStream.Index == streamIndex).Index;
+            if (streamIndex == null)
+            {
+                streamIndex = source.PrimaryVideoStream?.Index
```

Comment:
Maybe use `??=` instead of `if (streamIndex == null)...`

I.e.
```csharp
streamIndex ??= source.PrimaryVideoStream?.Index ?? source.VideoStreams.FirstOrDefault()?.Index ?? 0;
```